l10n_it_fatturapa_in/models/account.py

```diff
@@ -15,6 +17,44 @@ class AccountInvoice(models.Model):
         "einvoice.line", "invoice_id", string="Lines Detail",
         readonly=True, copy=False)
 
+    e_invoice_amount_untaxed = fields.Monetary(
+        string='E-Invoice Untaxed Amount', readonly=True)
+    e_invoice_amount_tax = fields.Monetary(string='E-Invoice Tax Amount',
+                                           readonly=True)
+    e_invoice_amount_total = fields.Monetary(string='E-Invoice Total Amount',
+                                             readonly=True)
+
+    @api.multi
+    def invoice_validate(self):
+        for invoice in self:
+            if (invoice.e_invoice_amount_untaxed and
+                    float_compare(invoice.amount_untaxed,
+                                  invoice.e_invoice_amount_untaxed,
+                                  precision_rounding=invoice.currency_id
+                                  .rounding) != 0):
+                raise ValidationError(
+                    _('Untaxed Amount %s differ from e-invoice %s') %
+                     (invoice.amount_untaxed,
+                      invoice.e_invoice_amount_untaxed))
```

Comment:
Non vorrei introdurre un controllo troppo rigido. Ci potrebbero essere casi in cui l'utente, consapevolmente, vuole comunque procedere a registrare la fattura.

Il controllo potrebbe essere fatto su richiesta, con un pulsante in fattura, a fianco al pulsante "valida".

Reply:
Si almeno inizialmente magari sarebbe preferibile non bloccare il flusso. Idealmente sarebbe utile poter notificare l'errore e chiedere all'utente se vuole continuare/forzare, però non saprei come impostare una logica di questo genere. I `button` hanno la `confirm` non fà proprio al caso nostro. 

Qualche idea oltre ad avere un pulsante separato?

Reply:
Si potrebbe mettere un avviso evidente, tipo questo:

![image](https://user-images.githubusercontent.com/1033131/55308154-21a8a100-545a-11e9-8673-8d93d7f913b3.png)

Reply:
> Si potrebbe mettere un avviso evidente, tipo questo:
> 
> ![image](https://user-images.githubusercontent.com/1033131/55308154-21a8a100-545a-11e9-8673-8d93d7f913b3.png)

Questo usa il widget `payment` per mostrare l'outstanding di una fattura, comunque dò uno sguardo per vedere come possiamo fare

Reply:
Non è un widget ma un normale div:
https://github.com/odoo/odoo/blob/acf7c1460afd169d0ab12155c68e0edd6abde027/addons/account/views/account_invoice_view.xml#L363-L366

Reply:
@eLBati Non l'avevo visto!!! Ok!

Reply:
@eLBati Ho appena push-ato la modifica